File: plugin/eval/fixtures/javascript-patterns/tree/src/ports.js
export function parsePort(text) {
  const value = Number(String(text).trim());
  if (!Number.isInteger(value)) {
    throw new RangeError(`not a port: ${text}`);
  }
  if (value < 1 || value > 65535) {
    throw new RangeError(`port out of range: ${value}`);
  }
  return value;
}

File: plugin/eval/fixtures/dotnet-patterns/tree/Ports.cs
namespace Ports;

/// <summary>Port parsing helpers.</summary>
public static class PortParser
{
    /// <summary>Parses <paramref name="text"/> as a TCP port in the range 1-65535.</summary>
    public static int Parse(string text)
    {
        if (!int.TryParse(text.Trim(), out var value))
        {
            throw new ArgumentException($"not a port: {text}", nameof(text));
        }
        if (value is < 1 or > 65535)
        {
            throw new ArgumentOutOfRangeException(nameof(text), value, "port out of range");
        }
        return value;
    }
}
